experimental/torch_xla2/examples/train_llama_torchtitan/train_llama.py

```diff
@@ -0,0 +1,283 @@
+import os
+import time
+import logging
+from typing import Tuple
+from collections import defaultdict
+import functools
+
+import torch
+import torch.nn.functional
+from torch.utils import _pytree as pytree
+
+import torch_xla2
+import torch_xla2.interop
+from torch_xla2.interop import jax_view, torch_view, JittableModule
+import jax
+import jax.numpy as jnp
+from jax.experimental import shard_map
+from jax.experimental import mesh_utils
+import optax
+
+from torchtitan.models.llama import llama3_configs
+from torchtitan.models.llama import model as titan
+
+P = jax.sharding.PartitionSpec
+
+def _setup_default_env():
+  os.environ.setdefault('TF_CPP_MIN_LOG_LEVEL', '1')
+  os.environ.setdefault('GRPC_VERBOSITY', 'ERROR')
+  os.environ.setdefault('ALLOW_MULTIPLE_LIBTPU_LOAD', '1')
+  # only need for tpu v4
+  # os.environ.setdefault('TPU_MEGACORE', 'megacore_dense')
+  tpu_args = "--xla_tpu_enable_async_collective_fusion_fuse_all_gather=true --xla_tpu_megacore_fusion_allow_ags=false --xla_enable_async_collective_permute=true --xla_tpu_enable_ag_backward_pipelining=true --xla_tpu_enable_data_parallel_all_reduce_opt=true --xla_tpu_data_parallel_opt_different_sized_ops=true --xla_tpu_enable_async_collective_fusion=true --xla_tpu_enable_async_collective_fusion_multiple_steps=true --xla_tpu_overlap_compute_collective_tc=true --xla_enable_async_all_gather=true"
+  os.environ.setdefault('LIBTPU_INIT_ARGS', tpu_args)
+
+_setup_default_env()
```

Comment:
Did you test that changing the environ here actually gets picked up by XLA? For example I was wondering if `import jax` or some other import will cause the TPU backend to get initialized and ignore future changes to XLA flags.

Reply:
hmmm... moving to the top to be safe.